### PR TITLE
[Tutorial] OpenVINOQuantizer

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
Fixes #ISSUE_NUMBER

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: prototype_source/openvino_quantizer.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+Introduction
+--------------
+
+This tutorial introduces the steps for utilizing the `Neural Network Compression Framework (nncf) <https://github.com/openvinotoolkit/nncf/tree/develop>`_ to generate a quantized model customized
+for the `OpenVINO torch.compile backend <https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html>`_ and explains how to lower the quantized model into the `OpenVINO <https://docs.openvino.ai/2024/index.html>`_ representation.
+
+The pytorch 2 export quantization flow uses the torch.export to capture the model into a graph and performs quantization transformations on top of the ATen graph.
+This approach is expected to have significantly higher model coverage, better programmability, and a simplified UX.
+OpenVINO is the new backend that compiles the FX Graph generated by TorchDynamo into an optimized OpenVINO model.
```
Comment (ynimmaga) [inline prototype_source/openvino_quantizer.rst:14]:
```suggestion
OpenVINO backend compiles the FX Graph generated by TorchDynamo into an optimized OpenVINO model.
```

File: prototype_source/openvino_quantizer.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+The quantization flow mainly includes four steps:
+
+- Step 1: Install OpenVINO and NNCF.
+- Step 2: Capture the FX Graph from the eager Model based on the `torch export mechanism <https://pytorch.org/docs/main/export.html>`_.
+- Step 3: Apply the Quantization flow based on the captured FX Graph.
+- Step 4: Lower the quantized model into OpenVINO representation with the API ``torch.compile``.
```
Comment (ynimmaga) [inline prototype_source/openvino_quantizer.rst:21]:
```suggestion
- Step 4: Lower the quantized model into OpenVINO representation with the API `torch.compile <https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html>`_.
```

File: prototype_source/openvino_quantizer.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    # Create the Eager Model
+    model_name = "resnet18"
+    model = models.__dict__[model_name](pretrained=True)
+
+    # Set the model to eval mode
+    model = model.eval()
+
+    # Create the data, using the dummy data here as an example
+    traced_bs = 50
+    x = torch.randn(traced_bs, 3, 224, 224).contiguous(memory_format=torch.channels_last)
```
Comment (ynimmaga) [inline prototype_source/openvino_quantizer.rst:97]: why do we need the memory format to be channels_last?

File: prototype_source/openvino_quantizer.rst [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    example_inputs = (x,)
+
+    # Capture the FX Graph to be quantized
+    with torch.no_grad(), disable_patching():
```
Comment (ynimmaga) [inline prototype_source/openvino_quantizer.rst:101]: is disable_patching() needed both during export and inference with torch.compile?

File: prototype_source/openvino_quantizer.rst [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+
+The optimized model is using low-level kernels designed specifically for Intel CPU.
+This should significantly speed up inference time in comparison with the eager model.
+
+Conclusion
+------------
+
+With this tutorial, we introduce how to use torch.compile with the OpenVINO backend and the OpenVINO quantizer.
+For further information, please visit `OpenVINO deploymet via torch.compile documentation <https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html>`_.
```
Comment (ynimmaga) [inline prototype_source/openvino_quantizer.rst:211]:
```suggestion
For further information, please visit `OpenVINO deployment via torch.compile documentation <https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html>`_.
```

File: prototype_source/openvino_quantizer.rst [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+3. Apply Quantization
+^^^^^^^^^^^^^^^^^^^^^^^
+
+After we capture the FX Module to be quantized, we will import the OpenVINOQuantizer.
+
+
+.. code-block:: python
+
+    quantizer = OpenVINOQuantizer()
```
Comment (alexsu52) [inline prototype_source/openvino_quantizer.rst:114]:
```suggestion
    from nncf.experimental.torch.fx.quantization.quantizer.openvino_quantizer import OpenVINOQuantizer

    quantizer = OpenVINOQuantizer()
```

File: prototype_source/openvino_quantizer.rst [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+        #Exclude by regular expression:
+        regex = '.*layer_.*'
+        OpenVINOQuantizer(ignored_scope=nncf.IgnoredScope(patterns=regex))
+
+        #Exclude by subgraphs:
+        # In this case, all nodes along all simple paths in the graph
+        # from input to output nodes will be excluded from the quantization process.
+        subgraph = nncf.Subgraph(inputs=['layer_1', 'layer_2'], outputs=['layer_3'])
+        OpenVINOQuantizer(ignored_scope=nncf.IgnoredScope(subgraphs=[subgraph]))
+
```
Comment (alexsu52) [inline prototype_source/openvino_quantizer.rst:157]: Where can I find more information about OpenVINOQuantizer parameters?

File: prototype_source/openvino_quantizer.rst [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+
+The optimized model is using low-level kernels designed specifically for Intel CPU.
+This should significantly speed up inference time in comparison with the eager model.
+
+Conclusion
+------------
+
+With this tutorial, we introduce how to use torch.compile with the OpenVINO backend and the OpenVINO quantizer.
```
Comment (alexsu52) [inline prototype_source/openvino_quantizer.rst:210]: I would suggest to add somethink like that:
For more information about NNCF and NNCF Quantization Flow for PyTorch models, please visit

File: prototype_source/openvino_quantizer.rst [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+    import copy
+    import openvino.torch
+    import torch
+    import torchvision.models as models
+    from torch.ao.quantization.quantize_pt2e import convert_pt2e
+    from torch.ao.quantization.quantize_pt2e import prepare_pt2e
+    from torch.ao.quantization.quantizer.openvino_quantizer import OpenVINOQuantizer
+
+    import nncf
+    from nncf.torch import disable_patching
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:86]: Unfortunately, that does not work. We can do `import nncf.torch` and then do `nncf.torch.disable_patching`

File: prototype_source/openvino_quantizer.rst (hunk 3 above)
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:97]: This is a copypast from the original tutorial, removed, thansk!

File: prototype_source/openvino_quantizer.rst (hunk 4 above)
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:101]: Unfortunately yes: export will fail with an error and performance of the compiled model will be ruined without it

File: prototype_source/openvino_quantizer.rst (hunk 8 above)
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst]: Done, please check

File: prototype_source/openvino_quantizer.rst [hunk 10]
```diff
@@ -0,0 +1,211 @@
+PyTorch 2 Export Quantization with OpenVINO backend
+===========================================================================
+
+**Author**: dlyakhov, asuslov, aamir, # TODO: add required authors
+
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:5]: Done

File: prototype_source/openvino_quantizer.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst]: `import nncf.torch` is introduced, please check

File: prototype_source/openvino_quantizer.rst (hunk 7 above)
Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:161]: I've added a link to nncf API docs, which should be updated with this PR: https://github.com/openvinotoolkit/nncf/pull/3277

File: prototype_source/openvino_quantizer.rst [hunk 11]
```diff
@@ -0,0 +1,245 @@
+PyTorch 2 Export Quantization for OpenVINO torch.compile backend.
+===========================================================================
+
+**Authors**: `Daniil Lyakhov <https://github.com/daniil-lyakhov>`_, `Alexander Suslov <https://github.com/alexsu52>`_, `Aamir Nazir <https://github.com/anzr299>`_
+
+Prerequisites
+--------------
+- `PyTorch 2 Export Post Training Quantization <https://pytorch.org/tutorials/prototype/pt2e_quant_ptq.html>`_
```
Comment (AlexKoff88) [inline prototype_source/openvino_quantizer.rst:8]:
```suggestion
- `PyTorch 2 Export Post-Training Quantization <https://pytorch.org/tutorials/prototype/pt2e_quant_ptq.html>`_
```

File: prototype_source/openvino_quantizer.rst [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+- `How to Write a Quantizer for PyTorch 2 Export Quantization <https://pytorch.org/tutorials/prototype/pt2e_quantizer.html>`_
+
+Introduction
+--------------
+
+This tutorial demonstrates how to use `OpenVINOQuantizer` from `Neural Network Compression Framework (NNCF) <https://github.com/openvinotoolkit/nncf/tree/develop>`_ in PyTorch 2 Export Quantization flow to generate a quantized model customized for the `OpenVINO torch.compile backend <https://docs.openvino.ai/2024/openvino-workflow/torch-compile.html>`_ and explains how to lower the quantized model into the `OpenVINO <https://docs.openvino.ai/2024/index.html>`_ representation.
```
Comment (AlexKoff88) [inline prototype_source/openvino_quantizer.rst:14]: It would be more attractive if to give the user an idea why it may need to use OpenVINOQuantizer (e.g. it is more accurate, performant, etc.)

Comment (daniil-lyakhov) [reply prototype_source/openvino_quantizer.rst:16]: Make sense! Description of advantages of OpenVINOQuantizer was added
